lib/std/math/gcd.zig

```diff
@@ -0,0 +1,48 @@
+//! Greatest common divisor (https://mathworld.wolfram.com/GreatestCommonDivisor.html)
+const std = @import("std");
+const expectEqual = std.testing.expectEqual;
+
+/// Returns the greatest common divisor (GCD) of two unsigned integers (a and b) which are not both zero.
+/// For example, the GCD of 8 and 12 is 4, that is, gcd(8, 12) == 4.
+pub fn gcd(a: anytype, b: anytype) @TypeOf(a, b) {
+
+    // only unsigned integers are allowed and not both must be zero
+    comptime switch (@typeInfo(@TypeOf(a, b))) {
+        .Int => |int| std.debug.assert(int.signedness == .unsigned),
+        .ComptimeInt => {
+            std.debug.assert(a >= 0);
+            std.debug.assert(b >= 0);
+            std.debug.assert(a != 0 or b != 0);
+        },
+        else => unreachable,
+    };
+
```

Comment:
Should probably still have a `std.debug.assert(a != 0 or b != 0);` just here. Since the above check would only apply in the case of two `comptime_int`s.

Reply:
Yeah, definitely. That should be moved outside.

Reply:
Yes, shure!